pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -614,6 +637,13 @@ func (gce *GCECloud) targetPoolURL(name, region string) string {
 	return fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/regions/%s/targetPools/%s", gce.projectID, region, name)
 }
 
+func (gce *GCECloud) GetLoadBalancerName(service *v1.Service) string {
+	//GCE requires that the name of a load balancer starts with a lower case letter.
+	ret := "a" + service.Namespace[0:13] + "-" + service.Name[0:13] + "-" + string(service.UID)
```

Comment:
This has to be less than 64.  UID is 36.  That leaves 27.  You're at 28 here.  I guess `[0:12]` for each segment?